src/core/ext/filters/client_channel/lb_policy/xds/xds_cluster_resolver.cc

```diff
@@ -444,26 +443,21 @@ void XdsClusterResolverLb::EdsDiscoveryMechanism::Orphan() {
 void XdsClusterResolverLb::LogicalDNSDiscoveryMechanism::Start() {
   std::string target =
       parent()->config_->discovery_mechanisms()[index()].dns_hostname;
-  grpc_channel_args* args = nullptr;
-  FakeResolverResponseGenerator* fake_resolver_response_generator =
-      grpc_channel_args_find_pointer<FakeResolverResponseGenerator>(
-          parent()->args_,
+  ChannelArgs args = parent()->args_;
+  auto* fake_resolver_response_generator =
+      args.GetPointer<FakeResolverResponseGenerator>(
```

Comment:
we're not loading this from the canonical name (instead loading it from `GRPC_ARG_XDS_LOGICAL_DNS_CLUSTER_FAKE_RESOLVER_RESPONSE_GENERATOR`) so this would not work